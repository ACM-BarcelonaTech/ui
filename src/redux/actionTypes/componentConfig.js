/**
 * Sets plot configs and data.
 */
const COMPONENT_CONFIG = 'componentConfig';
const LOAD_CONFIG = `${COMPONENT_CONFIG}/load`;
const UPDATE_CONFIG = `${COMPONENT_CONFIG}/update`;
const SAVE_CONFIG = `${COMPONENT_CONFIG}/save`;
const PLOT_DATA_LOADED = `${COMPONENT_CONFIG}/plotDataLoaded`;
const PLOT_DATA_LOADING = `${COMPONENT_CONFIG}/plotDataLoading`;

export {
  LOAD_CONFIG, UPDATE_CONFIG, SAVE_CONFIG,
  PLOT_DATA_LOADED, PLOT_DATA_LOADING,
};
