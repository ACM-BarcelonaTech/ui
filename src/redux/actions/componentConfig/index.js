import loadPlotConfig from './loadPlotConfig';
import loadComponentConfig from './loadComponentConfig';
import updatePlotConfig from './updatePlotConfig';
import savePlotConfig from './savePlotConfig';
import fetchPlotDataWork from './fetchPlotDataWork';

export {
  loadPlotConfig,
  loadComponentConfig,
  updatePlotConfig,
  savePlotConfig,
  fetchPlotDataWork,
};
