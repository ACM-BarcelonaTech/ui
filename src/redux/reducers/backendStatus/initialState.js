const initialExperimentBackendStatus = {
  loading: false,
  error: false,
  status: {
  },
};

const initialState = {};

export default initialState;

export { initialExperimentBackendStatus };
