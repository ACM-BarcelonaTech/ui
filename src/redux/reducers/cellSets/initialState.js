const initialState = {
  properties: {},
  hierarchy: [],
  selected: [],
  loading: true,
  error: false,
};

export default initialState;
