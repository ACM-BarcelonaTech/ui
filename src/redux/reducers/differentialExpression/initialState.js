const initialState = {
  properties: {
    data: [],
    comparisonGroup: {},
    comparisonType: null,
    loading: false,
    error: false,
    total: 0,
  },
  comparison: {
    type: 'within',
    advancedFilters: [],
    group: {
      between: {
        cellSet: null,
        compareWith: null,
        basis: null,
      },
      within: {
        cellSet: null,
        compareWith: null,
        basis: null,
      },
    },
    ordering: {
      orderBy: 'logFC',
      orderDirection: 'DESC',
    },
  },
};

export default initialState;
