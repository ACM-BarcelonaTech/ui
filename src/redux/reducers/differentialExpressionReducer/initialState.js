const initialState = {
  properties: {
    data: {},
    cellSets: {},
    loading: false,
    error: false,
  },
};

export default initialState;
