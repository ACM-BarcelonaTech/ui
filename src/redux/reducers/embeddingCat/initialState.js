const initialState = {
  width: 700,
  height: 550,
  pointSize: 5,
  toggleInvert: '#FFFFFF',
  masterColour: '#000000',
  umap1Domain: null,
  umap2Domain: null,
  titleText: '',
  titleSize: 20,
  titleAnchor: 'start',
  axisTitlesize: 13,
  axisTicks: 13,
  transGrid: 0,
  axesOffset: 10,
  masterFont: 'sans-serif',
  xaxisText: 'UMAP 1',
  yaxisText: 'UMAP 2',
  pointStyle: 'circle',
  pointOpa: 5,
  g1Color: 'red',
  g2mColor: 'green',
  sColor: 'blue',
  legendTextColor: '#FFFFFF',
  legendEnabled: true,
  legend: null,
  geneexpLegendloc: '',
  labelSize: 28,
  labelShow: 1,
  labelFont: 2,
  labelsEnabled: true,
  selectedClusters: [],
  testVar: null,
};

export default initialState;
