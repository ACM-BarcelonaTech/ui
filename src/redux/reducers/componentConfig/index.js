import {
  UPDATE_CONFIG, SAVE_CONFIG, LOAD_CONFIG,
  PLOT_DATA_LOADING, PLOT_DATA_ERROR,
} from 'redux/actionTypes/componentConfig';

import initialState from './initialState';
import loadConfig from './loadConfig';
import updateConfig from './updateConfig';
import saveConfig from './saveConfig';
import plotDataLoading from './plotDataLoading';
import plotDataError from './plotDataError';

const plotsReducer = (state = initialState, action) => {
  switch (action.type) {
    case LOAD_CONFIG:
      return loadConfig(state, action);
    case UPDATE_CONFIG:
      return updateConfig(state, action);
    case SAVE_CONFIG:
      return saveConfig(state, action);
    case PLOT_DATA_LOADING:
      return plotDataLoading(state, action);
    case PLOT_DATA_ERROR:
      return plotDataError(state, action);
    default:
      return state;
  }
};

export default plotsReducer;
