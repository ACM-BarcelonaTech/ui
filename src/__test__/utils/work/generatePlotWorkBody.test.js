import plotNames from 'utils/plots/plotNames';
import generatePlotWorkBody from 'utils/work/generatePlotWorkBody';

import { initialPlotConfigStates } from 'redux/reducers/componentConfig/initialState';

describe('Generate plot work request body', () => {
  it('Generates the correct work request body for DotPlot', () => {
    const workBody = generatePlotWorkBody(plotNames.DOT_PLOT, initialPlotConfigStates.dotPlot);

    expect(workBody.name).toEqual(plotNames.DOT_PLOT);
    expect(workBody).toMatchSnapshot();
  });
});
