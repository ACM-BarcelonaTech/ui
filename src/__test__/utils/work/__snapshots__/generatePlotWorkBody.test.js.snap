// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generate plot work request body Generates the correct work request body for DotPlot 1`] = `
Object {
  "input": Object {
    "genes": Array [],
    "nGenes": 3,
  },
  "markerGenes": false,
  "name": "DotPlot",
  "subset": Object {
    "cellClassKey": "louvain",
    "cellSetKey": "All",
  },
}
`;
